packages/core/admin/server/src/controllers/webhooks.ts

```diff
@@ -20,7 +21,27 @@ const urlRegex =
 const webhookValidator = yup
   .object({
     name: yup.string().required(),
-    url: yup.string().matches(urlRegex, 'url must be a valid URL').required(),
+    url: yup
+      .string()
+      .matches(urlRegex, 'url must be a valid URL')
+      .required()
+      .test(
+        'is-public-url',
+        "Url is not supported because it isn't reachable over the public internet",
+        async (url) => {
+          if (process.env.NODE_ENV !== 'production') {
+            return true;
+          }
+
+          try {
+            const parsedUrl = new URL(url!);
+            const isLocalUrl = await isLocalhostIp(parsedUrl.hostname);
```

Comment:
To be completely safe you need to use `punycode` on the url first since is-localhost-ip doesn't support international character sets by default.  Otherwise (in theory) someone could, for example, use an internationalized domain name pointed at 127.0.0.1 to achieve the same effect.